tests/unit/test_serializers.py

```diff
@@ -1,42 +1,110 @@
         assert (os.path.join('fake_dir', 'cassette_name.json') ==
-                JSONSerializer.generate_cassette_name(self.cassette_dir,
-                                                      self.cassette_name))
+                json_serializer.JSONSerializer.generate_cassette_name(
+                    self.cassette_dir,
+                    self.cassette_name,
+                ))
 
     def test_generate_cassette_name_with_instance(self):
-        serializer = JSONSerializer()
+        """Verify generate_cassette_name works on an instance too."""
+        serializer = json_serializer.JSONSerializer()
         assert (os.path.join('fake_dir', 'cassette_name.json') ==
                 serializer.generate_cassette_name(self.cassette_dir,
                                                   self.cassette_name))
 
 
-class Serializer(BaseSerializer):
+class Serializer(base.BaseSerializer):
+    """Serializer to test NotImplementedError exceptions."""
+
     name = 'test'
 
 
+class BytesSerializer(base.BaseSerializer):
+    """Serializer to test stored_as_binary."""
+
+    name = 'bytes-test'
+    stored_as_binary = True
+
+    # NOTE(sigmavirus24): These bytes, when decoded, result in a
+    # UnicodeDecodeError
+    serialized_bytes = b"hi \xAD"
+
+    def serialize(self, *args):
+        """Return the problematic bytes."""
+        return self.serialized_bytes
+
+    def deserialize(self, *args):
+        """Return the problematic bytes."""
+        return self.serialized_bytes
+
+
 class TestBaseSerializer(unittest.TestCase):
+    """Tests around BaseSerializer behaviour."""
+
     def test_serialize_is_an_interface(self):
+        """Verify we handle unimplemented methods."""
         serializer = Serializer()
         with pytest.raises(NotImplementedError):
             serializer.serialize({})
 
     def test_deserialize_is_an_interface(self):
+        """Verify we handle unimplemented methods."""
         serializer = Serializer()
         with pytest.raises(NotImplementedError):
             serializer.deserialize('path')
 
     def test_requires_a_name(self):
+        """Verify we handle unimplemented methods."""
         with pytest.raises(ValueError):
-            BaseSerializer()
+            base.BaseSerializer()
+
+
+class TestBinarySerializers(unittest.TestCase):
+    """Verify the behaviour of stored_as_binary."""
+
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        serializer = BytesSerializer()
+        self.cassette_path = 'test_cassette.test'
+        self.proxy = proxy.SerializerProxy(
+            serializer,
+            self.cassette_path,
+            allow_serialization=True,
+        )
+
+    def test_serialize(self):
+        """Verify we use the right mode with open()."""
+        mopen = mock.mock_open()
+        with mock.patch('betamax.serializers.proxy.open', mopen):
+            self.proxy.serialize({'cassette': 'data'})
+
+        assert mock.call(self.cassette_path, 'wb') in mopen.mock_calls
```

Comment:
@sigmavirus24 Should we also check that `'w'` is used when `stored_as_binary` is `False`, or am I being to paranoid?

Reply:
Good point. I also simplified this test significantly.